### PR TITLE
clients/horizonclient: make errors easier to understand at a glance

### PR DESCRIPTION
**Pull request by leighmcculloch**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [x] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [x] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Include the transaction result codes in the horizonclient.Error string that is outputted when serializing the error to a string.

### Why

When developing the transaction failed error is very opaque, displaying only `"Transaction Failed"` in logs. This is fine for code that has been written well to unwrap the error using `horizonclient.GetError`, and that also uses `ResultsCodes()`. But developers, even experienced ones, don't often start out writing code that inspects errors perfectly. It's helpful to get succinct textual errors that are as meaningful as possible, and really pin point what happened. Including the result code in the string that is output will in most cases significantly increase the signal a developer gets on why a transaction failed.

This is a recurring pain point for me using the horizonclient package in applications.

This function was not tested, so I added tests first in the first commit and the second commit contains the changes to the test and the new functionality.

### Known limitations

N/A

### CODE REVIEW

File: services/regulated-assets-approval-server/internal/serve/httperror/http_error_test.go [hunk 1]
```diff
@@ -35,5 +35,5 @@ func TestParseHorizonError(t *testing.T) {
 		},
 	}
 	err = ParseHorizonError(horizonError)
-	require.EqualError(t, err, "error submitting transaction: problem: bad_request, &{TransactionCode:tx_code_here OperationCodes:[op_success op_bad_auth]}\n: horizon error: \"Bad Request\" - check horizon.Error.Problem for more information")
+	require.EqualError(t, err, "error submitting transaction: problem: bad_request, &{TransactionCode:tx_code_here OperationCodes:[op_success op_bad_auth]}\n: horizon error: \"Bad Request\" (tx_code_here, op_success, op_bad_auth) - check horizon.Error.Problem for more information")
```
Comment (leighmcculloch) [inline services/regulated-assets-approval-server/internal/serve/httperror/http_error_test.go:38]: @marcelosalloum @Zagan202 I updated a test in the `regulated-assets-approval-server`. This change might simplify the logic you need in the `ParseHorizonError` function, but I've kept this PR hyper-focused on the change in the horizonclient and so only changed the test here minimally.

**Review by ire-and-curses — APPROVED** [review]
Looks great, thanks for this!

File: services/regulated-assets-approval-server/internal/serve/httperror/http_error_test.go (hunk 1 above)
Comment (Zagan202) [reply services/regulated-assets-approval-server/internal/serve/httperror/http_error_test.go:38]: Thanks for changing this I'll look into how we can simplify our `ParseHorizonError` logic after this is merged.

**Review by Zagan202 — APPROVED** [review]
LGTM

**Pull request merged**